[sqlancer] Unexpected exception: failed to init view stmt

### Steps to reproduce the behavior (Required)
1. Prepare Data:

```sql
-- Time: 2023/08/10 00:22:54
-- Database: v3
-- Database version: 5.1.0
-- seed value: 1691598037413
DROP DATABASE IF EXISTS v3;
CREATE DATABASE v3;
USE v3;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE t0 (c_0_0 DATETIME NULL ,c_0_1 BOOLEAN NULL ,c_0_2 DECIMAL(30, 0)  NULL ,c_0_3 DATETIME NOT NULL ,c_0_4 DECIMAL(7, 0)  NOT NULL ,c_0_5 DECIMAL(25, 14)  NULL ,c_0_6 DECIMAL(4, 2)  NULL ,c_0_7 TINYINT NULL ) UNIQUE KEY (c_0_0,c_0_1,c_0_2,c_0_3,c_0_4,c_0_5,c_0_6) PARTITION BY RANGE(c_0_0) (START ("2010-01-01") END ("2022-12-31") EVERY (INTERVAL 339 day)) DISTRIBUTED BY HASH (c_0_2,c_0_0) BUCKETS 3 properties("replication_num"="1") ;
CREATE TABLE IF NOT EXISTS t1 (c_1_0 VARCHAR(1) NOT NULL,c_1_1 VARCHAR(11) NOT NULL,c_1_2 DATE NOT NULL,c_1_3 DATETIME NOT NULL,c_1_4 BOOLEAN NOT NULL,c_1_5 VARCHAR(21) NOT NULL,c_1_6 BOOLEAN NOT NULL,c_1_7 DATETIME NOT NULL,c_1_8 INT NOT NULL,c_1_9 BOOLEAN NOT NULL,c_1_10 LARGEINT NOT NULL,c_1_11 DOUBLE NOT NULL,c_1_12 LARGEINT NOT NULL,c_1_13 DATETIME NOT NULL,c_1_14 DATETIME NOT NULL,c_1_15 BOOLEAN NOT NULL,c_1_16 INT NOT NULL,c_1_17 FLOAT NOT NULL,c_1_18 ARRAY<STRING> NOT NULL, INDEX index1_c_1_4(`c_1_4`) USING BITMAP) PRIMARY KEY (c_1_0,c_1_1,c_1_2,c_1_3,c_1_4,c_1_5,c_1_6,c_1_7,c_1_8,c_1_9,c_1_10) PARTITION BY RANGE(c_1_2) (START ("2010-01-01") END ("2022-12-31") EVERY (INTERVAL 98 day)) DISTRIBUTED BY HASH (c_1_1,c_1_4,c_1_6,c_1_10) BUCKETS 3 properties("replication_num"="1") ;
CREATE MATERIALIZED VIEW v0 AS SELECT t1_1154.c_1_0, t1_1154.c_1_1, t1_1154.c_1_2, t1_1154.c_1_3, t1_1154.c_1_4, t1_1154.c_1_5, t1_1154.c_1_6, t1_1154.c_1_7, t1_1154.c_1_8, t1_1154.c_1_9, t1_1154.c_1_10, t1_1154.c_1_11, t1_1154.c_1_12, t1_1154.c_1_13, t1_1154.c_1_14, t1_1154.c_1_16, t1_1154.c_1_17, MIN(t1_1154.c_1_15)  FROM t1 AS t1_1154 GROUP BY t1_1154.c_1_0, t1_1154.c_1_1, t1_1154.c_1_2, t1_1154.c_1_3, t1_1154.c_1_4, t1_1154.c_1_5, t1_1154.c_1_6, t1_1154.c_1_7, t1_1154.c_1_8, t1_1154.c_1_9, t1_1154.c_1_10, t1_1154.c_1_11, t1_1154.c_1_12, t1_1154.c_1_13, t1_1154.c_1_14, t1_1154.c_1_16, t1_1154.c_1_17 ORDER BY t1_1154.c_1_0, t1_1154.c_1_1 ASC, t1_1154.c_1_2 ASC, t1_1154.c_1_3, t1_1154.c_1_4 DESC, t1_1154.c_1_5 ASC, t1_1154.c_1_6, t1_1154.c_1_7, t1_1154.c_1_8 DESC, t1_1154.c_1_9 DESC, t1_1154.c_1_10 DESC, t1_1154.c_1_11, t1_1154.c_1_12 ASC, t1_1154.c_1_13 ASC, t1_1154.c_1_14 ASC, t1_1154.c_1_16, t1_1154.c_1_17;
CREATE MATERIALIZED VIEW v0 AS SELECT t1_1156.c_1_1, t1_1156.c_1_5, t1_1156.c_1_6, t1_1156.c_1_8, t1_1156.c_1_10, t1_1156.c_1_11, t1_1156.c_1_12, t1_1156.c_1_13, t1_1156.c_1_16, MAX(t1_1156.c_1_0) , MAX(t1_1156.c_1_2) , MIN(t1_1156.c_1_3) , MIN(t1_1156.c_1_4) , MIN(t1_1156.c_1_7) , MIN(t1_1156.c_1_9) , MIN(t1_1156.c_1_14) , MAX(t1_1156.c_1_15) , MAX(t1_1156.c_1_17)  FROM t1 AS t1_1156 GROUP BY t1_1156.c_1_1, t1_1156.c_1_5, t1_1156.c_1_6, t1_1156.c_1_8, t1_1156.c_1_10, t1_1156.c_1_11, t1_1156.c_1_12, t1_1156.c_1_13, t1_1156.c_1_16 ORDER BY t1_1156.c_1_1, t1_1156.c_1_5 ASC, t1_1156.c_1_6 DESC, t1_1156.c_1_8 DESC, t1_1156.c_1_10 ASC, t1_1156.c_1_11 ASC, t1_1156.c_1_12, t1_1156.c_1_13 ASC, t1_1156.c_1_16;
TRUNCATE TABLE t0;
CREATE VIEW v29 AS WITH with_t_0 as (SELECT t0_1159.c_0_0, t0_1159.c_0_1, t0_1159.c_0_2, t0_1159.c_0_3, t0_1159.c_0_4, t0_1159.c_0_5, t0_1159.c_0_6, t0_1159.c_0_7 FROM t0 AS t0_1159) SELECT t0_1159.c_0_0 FROM t0 AS t0_1159, t1 AS t1_1160, with_t_0 AS with_t_0;
CREATE VIEW v0 AS WITH with_t_0 as (SELECT t0_1159.c_0_0, t0_1159.c_0_1, t0_1159.c_0_2, t0_1159.c_0_3, t0_1159.c_0_4, t0_1159.c_0_5, t0_1159.c_0_6, t0_1159.c_0_7 FROM t0 AS t0_1159) SELECT t0_1159.c_0_0 FROM t0 AS t0_1159, t1 AS t1_1160, with_t_0 AS with_t_0 WHERE true;
CREATE VIEW v1 AS WITH with_t_0 as (SELECT t0_1159.c_0_0, t0_1159.c_0_1, t0_1159.c_0_2, t0_1159.c_0_3, t0_1159.c_0_4, t0_1159.c_0_5, t0_1159.c_0_6, t0_1159.c_0_7 FROM t0 AS t0_1159) SELECT t0_1159.c_0_0 FROM t0 AS t0_1159, t1 AS t1_1160, with_t_0 AS with_t_0 WHERE (NOT (true));
CREATE VIEW v2 AS WITH with_t_0 as (SELECT t0_1159.c_0_0, t0_1159.c_0_1, t0_1159.c_0_2, t0_1159.c_0_3, t0_1159.c_0_4, t0_1159.c_0_5, t0_1159.c_0_6, t0_1159.c_0_7 FROM t0 AS t0_1159) SELECT t0_1159.c_0_0 FROM t0 AS t0_1159, t1 AS t1_1160, with_t_0 AS with_t_0 WHERE ((true) IS NULL);
CREATE VIEW v78 AS (WITH with_t_0 as (SELECT t0_1159.c_0_0, t0_1159.c_0_1, t0_1159.c_0_2, t0_1159.c_0_3, t0_1159.c_0_4, t0_1159.c_0_5, t0_1159.c_0_6, t0_1159.c_0_7 FROM t0 AS t0_1159) (SELECT t0_1159.c_0_0 FROM t0 AS t0_1159, t1 AS t1_1160, with_t_0 AS with_t_0 WHERE true) UNION ALL (SELECT t0_1159.c_0_0 FROM t0 AS t0_1159, t1 AS t1_1160, with_t_0 AS with_t_0 WHERE (NOT (true))) UNION ALL (SELECT t0_1159.c_0_0 FROM t0 AS t0_1159, t1 AS t1_1160, with_t_0 AS with_t_0 WHERE ((true) IS NULL)) );
CREATE VIEW v3 AS SELECT v78_1185.c_0_0 FROM v78 AS v78_1185, t1 AS t1_1180, v29 AS v29_1184, v1 AS v1_1182, t0 AS t0_1179, v0 AS v0_1181;
CREATE VIEW v4 AS SELECT v78_1185.c_0_0 FROM v78 AS v78_1185, t1 AS t1_1180, v29 AS v29_1184, v1 AS v1_1182, t0 AS t0_1179, v0 AS v0_1181 WHERE NULL;
CREATE VIEW v5 AS SELECT v78_1185.c_0_0 FROM v78 AS v78_1185, t1 AS t1_1180, v29 AS v29_1184, v1 AS v1_1182, t0 AS t0_1179, v0 AS v0_1181 WHERE (NOT (NULL));
CREATE VIEW v6 AS SELECT v78_1185.c_0_0 FROM v78 AS v78_1185, t1 AS t1_1180, v29 AS v29_1184, v1 AS v1_1182, t0 AS t0_1179, v0 AS v0_1181 WHERE ((NULL) IS NULL);
CREATE VIEW v95 AS ((SELECT v78_1185.c_0_0 FROM v78 AS v78_1185, t1 AS t1_1180, v29 AS v29_1184, v1 AS v1_1182, t0 AS t0_1179, v0 AS v0_1181 WHERE NULL) UNION ALL (SELECT v78_1185.c_0_0 FROM v78 AS v78_1185, t1 AS t1_1180, v29 AS v29_1184, v1 AS v1_1182, t0 AS t0_1179, v0 AS v0_1181 WHERE (NOT (NULL))) UNION ALL (SELECT v78_1185.c_0_0 FROM v78 AS v78_1185, t1 AS t1_1180, v29 AS v29_1184, v1 AS v1_1182, t0 AS t0_1179, v0 AS v0_1181 WHERE ((NULL) IS NULL)) );
CREATE VIEW v7 AS SELECT (('')||(NULL)) FROM  (SELECT t1_1225.c_1_0, t1_1225.c_1_1, t1_1225.c_1_2, t1_1225.c_1_3, t1_1225.c_1_4, t1_1225.c_1_5, t1_1225.c_1_6, t1_1225.c_1_7, t1_1225.c_1_8, t1_1225.c_1_9, t1_1225.c_1_10, t1_1225.c_1_11, t1_1225.c_1_12, t1_1225.c_1_13, t1_1225.c_1_14, t1_1225.c_1_15, t1_1225.c_1_16, t1_1225.c_1_17 FROM t1 AS t1_1225 WHERE false) subt1 LEFT SEMI JOIN (SELECT v0_1226.c_0_0 FROM v0 AS v0_1226 WHERE CASE  WHEN NULL THEN (NOT (true))  WHEN false THEN  ( v0_1226.c_0_0 ) <= ( v0_1226.c_0_0 )   WHEN ((v0_1226.c_0_0) IS NULL) THEN ((v0_1226.c_0_0) IS NULL) ELSE ((v0_1226.c_0_0) IS NULL) END) subv0 ON subt1.c_1_1 = subv0.c_0_0 AND subt1.c_1_1 < subv0.c_0_0 AND subt1.c_1_17 < subv0.c_0_0 AND subt1.c_1_17 = subv0.c_0_0;
CREATE VIEW v8 AS SELECT (('')||(NULL)) FROM  (SELECT t1_1225.c_1_0, t1_1225.c_1_1, t1_1225.c_1_2, t1_1225.c_1_3, t1_1225.c_1_4, t1_1225.c_1_5, t1_1225.c_1_6, t1_1225.c_1_7, t1_1225.c_1_8, t1_1225.c_1_9, t1_1225.c_1_10, t1_1225.c_1_11, t1_1225.c_1_12, t1_1225.c_1_13, t1_1225.c_1_14, t1_1225.c_1_15, t1_1225.c_1_16, t1_1225.c_1_17 FROM t1 AS t1_1225 WHERE false) subt1 LEFT SEMI JOIN (SELECT v0_1226.c_0_0 FROM v0 AS v0_1226 WHERE CASE  WHEN NULL THEN (NOT (true))  WHEN false THEN  ( v0_1226.c_0_0 ) <= ( v0_1226.c_0_0 )   WHEN ((v0_1226.c_0_0) IS NULL) THEN ((v0_1226.c_0_0) IS NULL) ELSE ((v0_1226.c_0_0) IS NULL) END) subv0 ON subt1.c_1_1 = subv0.c_0_0 AND subt1.c_1_1 < subv0.c_0_0 AND subt1.c_1_17 < subv0.c_0_0 AND subt1.c_1_17 = subv0.c_0_0 WHERE false;
CREATE VIEW v9 AS SELECT (('')||(NULL)) FROM  (SELECT t1_1225.c_1_0, t1_1225.c_1_1, t1_1225.c_1_2, t1_1225.c_1_3, t1_1225.c_1_4, t1_1225.c_1_5, t1_1225.c_1_6, t1_1225.c_1_7, t1_1225.c_1_8, t1_1225.c_1_9, t1_1225.c_1_10, t1_1225.c_1_11, t1_1225.c_1_12, t1_1225.c_1_13, t1_1225.c_1_14, t1_1225.c_1_15, t1_1225.c_1_16, t1_1225.c_1_17 FROM t1 AS t1_1225 WHERE false) subt1 LEFT SEMI JOIN (SELECT v0_1226.c_0_0 FROM v0 AS v0_1226 WHERE CASE  WHEN NULL THEN (NOT (true))  WHEN false THEN  ( v0_1226.c_0_0 ) <= ( v0_1226.c_0_0 )   WHEN ((v0_1226.c_0_0) IS NULL) THEN ((v0_1226.c_0_0) IS NULL) ELSE ((v0_1226.c_0_0) IS NULL) END) subv0 ON subt1.c_1_1 = subv0.c_0_0 AND subt1.c_1_1 < subv0.c_0_0 AND subt1.c_1_17 < subv0.c_0_0 AND subt1.c_1_17 = subv0.c_0_0 WHERE (NOT (false));
CREATE VIEW v10 AS SELECT (('')||(NULL)) FROM  (SELECT t1_1225.c_1_0, t1_1225.c_1_1, t1_1225.c_1_2, t1_1225.c_1_3, t1_1225.c_1_4, t1_1225.c_1_5, t1_1225.c_1_6, t1_1225.c_1_7, t1_1225.c_1_8, t1_1225.c_1_9, t1_1225.c_1_10, t1_1225.c_1_11, t1_1225.c_1_12, t1_1225.c_1_13, t1_1225.c_1_14, t1_1225.c_1_15, t1_1225.c_1_16, t1_1225.c_1_17 FROM t1 AS t1_1225 WHERE false) subt1 LEFT SEMI JOIN (SELECT v0_1226.c_0_0 FROM v0 AS v0_1226 WHERE CASE  WHEN NULL THEN (NOT (true))  WHEN false THEN  ( v0_1226.c_0_0 ) <= ( v0_1226.c_0_0 )   WHEN ((v0_1226.c_0_0) IS NULL) THEN ((v0_1226.c_0_0) IS NULL) ELSE ((v0_1226.c_0_0) IS NULL) END) subv0 ON subt1.c_1_1 = subv0.c_0_0 AND subt1.c_1_1 < subv0.c_0_0 AND subt1.c_1_17 < subv0.c_0_0 AND subt1.c_1_17 = subv0.c_0_0 WHERE ((false) IS NULL);
CREATE VIEW v11 AS ((SELECT (('')||(NULL)) FROM  (SELECT t1_1225.c_1_0, t1_1225.c_1_1, t1_1225.c_1_2, t1_1225.c_1_3, t1_1225.c_1_4, t1_1225.c_1_5, t1_1225.c_1_6, t1_1225.c_1_7, t1_1225.c_1_8, t1_1225.c_1_9, t1_1225.c_1_10, t1_1225.c_1_11, t1_1225.c_1_12, t1_1225.c_1_13, t1_1225.c_1_14, t1_1225.c_1_15, t1_1225.c_1_16, t1_1225.c_1_17 FROM t1 AS t1_1225 WHERE false) subt1 LEFT SEMI JOIN (SELECT v0_1226.c_0_0 FROM v0 AS v0_1226 WHERE CASE  WHEN NULL THEN (NOT (true))  WHEN false THEN  ( v0_1226.c_0_0 ) <= ( v0_1226.c_0_0 )   WHEN ((v0_1226.c_0_0) IS NULL) THEN ((v0_1226.c_0_0) IS NULL) ELSE ((v0_1226.c_0_0) IS NULL) END) subv0 ON subt1.c_1_1 = subv0.c_0_0 AND subt1.c_1_1 < subv0.c_0_0 AND subt1.c_1_17 < subv0.c_0_0 AND subt1.c_1_17 = subv0.c_0_0 WHERE false) UNION ALL (SELECT (('')||(NULL)) FROM  (SELECT t1_1225.c_1_0, t1_1225.c_1_1, t1_1225.c_1_2, t1_1225.c_1_3, t1_1225.c_1_4, t1_1225.c_1_5, t1_1225.c_1_6, t1_1225.c_1_7, t1_1225.c_1_8, t1_1225.c_1_9, t1_1225.c_1_10, t1_1225.c_1_11, t1_1225.c_1_12, t1_1225.c_1_13, t1_1225.c_1_14, t1_1225.c_1_15, t1_1225.c_1_16, t1_1225.c_1_17 FROM t1 AS t1_1225 WHERE false) subt1 LEFT SEMI JOIN (SELECT v0_1226.c_0_0 FROM v0 AS v0_1226 WHERE CASE  WHEN NULL THEN (NOT (true))  WHEN false THEN  ( v0_1226.c_0_0 ) <= ( v0_1226.c_0_0 )   WHEN ((v0_1226.c_0_0) IS NULL) THEN ((v0_1226.c_0_0) IS NULL) ELSE ((v0_1226.c_0_0) IS NULL) END) subv0 ON subt1.c_1_1 = subv0.c_0_0 AND subt1.c_1_1 < subv0.c_0_0 AND subt1.c_1_17 < subv0.c_0_0 AND subt1.c_1_17 = subv0.c_0_0 WHERE (NOT (false))) UNION ALL (SELECT (('')||(NULL)) FROM  (SELECT t1_1225.c_1_0, t1_1225.c_1_1, t1_1225.c_1_2, t1_1225.c_1_3, t1_1225.c_1_4, t1_1225.c_1_5, t1_1225.c_1_6, t1_1225.c_1_7, t1_1225.c_1_8, t1_1225.c_1_9, t1_1225.c_1_10, t1_1225.c_1_11, t1_1225.c_1_12, t1_1225.c_1_13, t1_1225.c_1_14, t1_1225.c_1_15, t1_1225.c_1_16, t1_1225.c_1_17 FROM t1 AS t1_1225 WHERE false) subt1 LEFT SEMI JOIN (SELECT v0_1226.c_0_0 FROM v0 AS v0_1226 WHERE CASE  WHEN NULL THEN (NOT (true))  WHEN false THEN  ( v0_1226.c_0_0 ) <= ( v0_1226.c_0_0 )   WHEN ((v0_1226.c_0_0) IS NULL) THEN ((v0_1226.c_0_0) IS NULL) ELSE ((v0_1226.c_0_0) IS NULL) END) subv0 ON subt1.c_1_1 = subv0.c_0_0 AND subt1.c_1_1 < subv0.c_0_0 AND subt1.c_1_17 < subv0.c_0_0 AND subt1.c_1_17 = subv0.c_0_0 WHERE ((false) IS NULL)) );
CREATE VIEW v12 AS SELECT v10_1298.concat('', NULL) FROM v10 AS v10_1298, v2 AS v2_1300, v1 AS v1_1297, v29 AS v29_1301, v7 AS v7_1306, v8 AS v8_1308, v6 AS v6_1305, t0 AS t0_1294, v11 AS v11_1299, v4 AS v4_1303, v0 AS v0_1296, v3 AS v3_1302, v95 AS v95_1310, t1 AS t1_1295;

```

2. Query

```sql
CREATE VIEW v12 AS SELECT v10_1298.concat('', NULL) FROM v10 AS v10_1298, v2 AS v2_1300, v1 AS v1_1297, v29 AS v29_1301, v7 AS v7_1306, v8 AS v8_1308, v6 AS v6_1305, t0 AS t0_1294, v11 AS v11_1299, v4 AS v4_1303, v0 AS v0_1296, v3 AS v3_1302, v95 AS v95_1310, t1 AS t1_1295;
```
### Expected behavior (Required)
NO Error

### Real behavior (Required)
Error as follows
```
--java.lang.AssertionError: CREATE VIEW v12 AS SELECT v10_1298.concat('', NULL) FROM v10 AS v10_1298, v2 AS v2_1300, v1 AS v1_1297, v29 AS v29_1301, v7 AS v7_1306, v8 AS v8_1308, v6 AS v6_1305, t0 AS t0_1294, v11 AS v11_1299, v4 AS v4_1303, v0 AS v0_1296, v3 AS v3_1302, v95 AS v95_1310, t1 AS t1_1295;
--	at sqlancer.common.query.QueryAdapter.checkException(QueryAdapter.java:98)
--	at sqlancer.common.query.QueryAdapter.execute(QueryAdapter.java:91)
--	at sqlancer.Main$QueryManager.execute(Main.java:257)
--	at sqlancer.GlobalState.executeStatement(GlobalState.java:131)
--	at sqlancer.starrocks.oracle.tlp.StarRocksTLPViewOracle.createView(StarRocksTLPViewOracle.java:108)
--	at sqlancer.starrocks.oracle.tlp.StarRocksTLPViewOracle.check(StarRocksTLPViewOracle.java:33)
--	at sqlancer.ProviderAdapter.generateAndTestDatabase(ProviderAdapter.java:55)
--	at sqlancer.Main$DBMSExecutor.run(Main.java:342)
--	at sqlancer.Main$2.run(Main.java:557)
--	at sqlancer.Main$2.runThread(Main.java:533)
--	at sqlancer.Main$2.run(Main.java:523)
--	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
--	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
--	at java.lang.Thread.run(Thread.java:748)
--Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Unexpected exception: failed to init view stmt
--	at sun.reflect.GeneratedConstructorAccessor6.newInstance(Unknown Source)
--	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
--	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
--	at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)
--	at com.mysql.jdbc.Util.getInstance(Util.java:408)
--	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:943)
--	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3973)
--	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3909)
--	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2527)
--	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2680)
--	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2497)
--	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2455)
--	at com.mysql.jdbc.StatementImpl.executeInternal(StatementImpl.java:839)
--	at com.mysql.jdbc.StatementImpl.execute(StatementImpl.java:739)
--	at sqlancer.common.query.QueryAdapter.execute(QueryAdapter.java:85)
--	... 12 more
--
```
### StarRocks version (Required)
- main(b0cfdcf)

--- a/fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AstToSQLBuilder.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AstToSQLBuilder.java
@@ -15,7 +15,10 @@
 package com.starrocks.sql.analyzer;
 
 import com.google.common.base.Joiner;
-import com.starrocks.analysis.*;
+import com.starrocks.analysis.Expr;
+import com.starrocks.analysis.ParseNode;
+import com.starrocks.analysis.SlotRef;
+import com.starrocks.analysis.TableName;
 import com.starrocks.catalog.Table;
 import com.starrocks.catalog.Type;
 import com.starrocks.common.util.ParseUtil;

--- a/fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AstToSQLBuilder.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AstToSQLBuilder.java
@@ -15,10 +15,7 @@
 package com.starrocks.sql.analyzer;
 
 import com.google.common.base.Joiner;
-import com.starrocks.analysis.Expr;
-import com.starrocks.analysis.ParseNode;
-import com.starrocks.analysis.SlotRef;
-import com.starrocks.analysis.TableName;
+import com.starrocks.analysis.*;
 import com.starrocks.catalog.Table;
 import com.starrocks.catalog.Type;
 import com.starrocks.common.util.ParseUtil;
@@ -40,6 +37,7 @@ import org.apache.commons.collections4.CollectionUtils;
 import org.apache.commons.lang3.StringUtils;
 
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.stream.Collectors;
@@ -142,6 +140,17 @@ public class AstToSQLBuilder {
                 sqlBuilder.append("DISTINCT ");
             }
 
+            Map<FunctionCallExpr, String> selectFuncCallExprAliasMap = new HashMap<>();
+            for (SelectListItem selectListItem : selectList.getItems()) {
+                if (selectListItem.getExpr() != null
+                        && !(selectListItem.getExpr() instanceof FieldReference)
+                        && !(selectListItem.getExpr() instanceof SlotRef)
+                ) {
+                    selectFuncCallExprAliasMap.put((FunctionCallExpr)selectListItem.getExpr(),
+                            selectListItem.getAlias());
+                }
+            }
+
             List<String> selectListString = new ArrayList<>();
             if (CollectionUtils.isNotEmpty(stmt.getOutputExpression())) {
                 for (int i = 0; i < stmt.getOutputExpression().size(); ++i) {
@@ -161,7 +170,7 @@ public class AstToSQLBuilder {
                                     columnName));
                         }
                     } else {
-                        String aliasColumn = stmt.getSelectList().getItems().get(i).getAlias();
+                        String aliasColumn = selectFuncCallExprAliasMap.get(expr);
                         selectListString.add(
                                 aliasColumn == null ?
                                         visit(expr) : visit(expr) + " AS `" + aliasColumn + "`");

--- a/fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AstToSQLBuilder.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AstToSQLBuilder.java
@@ -85,7 +85,9 @@ public class AstToSQLBuilder {
                 res += ".";
             }
 
-            res += '`' + fieldName + '`';
+            res +=  fieldName.startsWith("`") ?
+                    fieldName
+                    : '`' + fieldName + '`';
             if (!fieldName.equalsIgnoreCase(columnName)) {
                 res += " AS `" + columnName + "`";
             }
@@ -163,7 +165,7 @@ public class AstToSQLBuilder {
                     } else {
                         selectListString.add(
                                 expr.getFn() == null || expr.getFn().getFunctionName().getDb() == null ?
-                                        visit(expr) :
+                                        visit(expr) + " AS `" + columnName.replace("`", "") + "`" :
                                         visit(expr) + " AS `" + expr.getFn().getFunctionName().getFunction() + "`");
                     }
                 }

--- a/fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AstToSQLBuilder.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AstToSQLBuilder.java
@@ -161,10 +161,10 @@ public class AstToSQLBuilder {
                                     columnName));
                         }
                     } else {
+                        String aliasColumn = stmt.getSelectList().getItems().get(i).getAlias();
                         selectListString.add(
-                                expr.getFn() == null || expr.getFn().getFunctionName().getDb() == null ?
-                                        visit(expr) + " AS `" + columnName + "`" :
-                                        visit(expr) + " AS `" + expr.getFn().getFunctionName().getFunction() + "`");
+                                aliasColumn == null ?
+                                        visit(expr) : visit(expr) + " AS `" + aliasColumn + "`");
                     }
                 }
             } else {

--- a/fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AstToSQLBuilder.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AstToSQLBuilder.java
@@ -140,13 +140,13 @@ public class AstToSQLBuilder {
                 sqlBuilder.append("DISTINCT ");
             }
 
-            Map<FunctionCallExpr, String> selectFuncCallExprAliasMap = new HashMap<>();
+            Map<Expr, String> selectExprAliasMap = new HashMap<>();
             for (SelectListItem selectListItem : selectList.getItems()) {
                 if (selectListItem.getExpr() != null
                         && !(selectListItem.getExpr() instanceof FieldReference)
                         && !(selectListItem.getExpr() instanceof SlotRef)
                 ) {
-                    selectFuncCallExprAliasMap.put((FunctionCallExpr)selectListItem.getExpr(),
+                    selectExprAliasMap.put(selectListItem.getExpr(),
                             selectListItem.getAlias());
                 }
             }
@@ -170,7 +170,7 @@ public class AstToSQLBuilder {
                                     columnName));
                         }
                     } else {
-                        String aliasColumn = selectFuncCallExprAliasMap.get(expr);
+                        String aliasColumn = selectExprAliasMap.get(expr);
                         selectListString.add(
                                 aliasColumn == null ?
                                         visit(expr) : visit(expr) + " AS `" + aliasColumn + "`");

--- a/fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AstToSQLBuilder.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AstToSQLBuilder.java
@@ -40,7 +40,6 @@ import org.apache.commons.collections4.CollectionUtils;
 import org.apache.commons.lang3.StringUtils;
 
 import java.util.ArrayList;
-import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.stream.Collectors;

--- a/fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AstToSQLBuilder.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AstToSQLBuilder.java
@@ -143,17 +143,6 @@ public class AstToSQLBuilder {
                 sqlBuilder.append("DISTINCT ");
             }
 
-            Map<Expr, String> selectExprAliasMap = new HashMap<>();
-            for (SelectListItem selectListItem : selectList.getItems()) {
-                if (selectListItem.getExpr() != null
-                        && !(selectListItem.getExpr() instanceof FieldReference)
-                        && !(selectListItem.getExpr() instanceof SlotRef)
-                ) {
-                    selectExprAliasMap.put(selectListItem.getExpr(),
-                            selectListItem.getAlias());
-                }
-            }
-
             List<String> selectListString = new ArrayList<>();
             if (CollectionUtils.isNotEmpty(stmt.getOutputExpression())) {
                 for (int i = 0; i < stmt.getOutputExpression().size(); ++i) {
@@ -173,10 +162,10 @@ public class AstToSQLBuilder {
                                     columnName));
                         }
                     } else {
-                        String aliasColumn = selectExprAliasMap.get(expr);
                         selectListString.add(
-                                aliasColumn == null ?
-                                        visit(expr) : visit(expr) + " AS `" + aliasColumn + "`");
+                                expr.getFn() == null || expr.getFn().getFunctionName().getDb() == null ?
+                                        visit(expr) :
+                                        visit(expr) + " AS `" + expr.getFn().getFunctionName().getFunction() + "`");
                     }
                 }
             } else {

--- a/fe/fe-core/src/test/java/com/starrocks/analysis/ShowCreateViewStmtTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/analysis/ShowCreateViewStmtTest.java
@@ -150,15 +150,13 @@ public class ShowCreateViewStmtTest {
         testCases.add(new String[]{"test_view_0",
                 "create view test_view_0 AS SELECT " +
                         " *, concat('', null) FROM `test`.`tbl1`",
-                "CREATE VIEW `test_view_0` (`k1`, `k2`, `v1`, `concat('', NULL)`) " +
-                        "AS SELECT `test`.`tbl1`.`k1`, `test`.`tbl1`.`k2`, `test`.`tbl1`.`v1`, concat('', NULL)\n" +
+                "CREATE VIEW `test_view_0` (`k1`, `k2`, `v1`, `concat('', NULL)`) AS SELECT `test`.`tbl1`.`k1`, `test`.`tbl1`.`k2`, `test`.`tbl1`.`v1`, concat('', NULL) AS `concat('', NULL)`\n" +
                         "FROM `test`.`tbl1`;"
         });
         testCases.add(new String[]{"test_view_1",
                 "create view test_view_1 AS SELECT " +
                         "concat(`test`.`tbl1`.`k1`, `test`.`tbl1`.`k2`) FROM `test`.`tbl1`",
-                "CREATE VIEW `test_view_1` (`concat(test.tbl1.k1, test.tbl1.k2)`) " +
-                        "AS SELECT concat(`test`.`tbl1`.`k1`, `test`.`tbl1`.`k2`)\n" +
+                "CREATE VIEW `test_view_1` (`concat(test.tbl1.k1, test.tbl1.k2)`) AS SELECT concat(`test`.`tbl1`.`k1`, `test`.`tbl1`.`k2`) AS `concat(test.tbl1.k1, test.tbl1.k2)`\n" +
                         "FROM `test`.`tbl1`;"
         });
         testCases.add(new String[]{"test_view_2",

--- a/fe/fe-core/src/test/java/com/starrocks/analysis/ShowCreateViewStmtTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/analysis/ShowCreateViewStmtTest.java
@@ -38,6 +38,7 @@ import org.junit.Assert;
 import org.junit.BeforeClass;
 import org.junit.Test;
 
+import java.util.ArrayList;
 import java.util.List;
 
 import static com.starrocks.sql.optimizer.statistics.CachedStatisticStorageTest.DEFAULT_CREATE_TABLE_TEMPLATE;
@@ -140,6 +141,55 @@ public class ShowCreateViewStmtTest {
             GlobalStateMgr.getCurrentState().dropTable(dropTableStmt);
         } catch (Exception ex) {
 
+        }
+    }
+
+    @Test
+    public void testCreateView() throws Exception {
+        List<String[]> testCases = new ArrayList<>();
+        testCases.add(new String[]{"test_view_1",
+                "create view test_view_1 AS SELECT " +
+                        "concat(`test`.`tbl1`.`k1`, `test`.`tbl1`.`k2`) FROM `test`.`tbl1`",
+                "CREATE VIEW `test_view_1` (`concat(test.tbl1.k1, test.tbl1.k2)`) " +
+                        "AS SELECT concat(`test`.`tbl1`.`k1`, `test`.`tbl1`.`k2`)\n" +
+                        "FROM `test`.`tbl1`;"
+        });
+        testCases.add(new String[]{"test_view_2",
+                "create view test_view_2 AS SELECT " +
+                        "`test`.`tbl1`.`k1`, `test`.`tbl1`.`k2` FROM `test`.`tbl1`",
+                "CREATE VIEW `test_view_2` (`k1`, `k2`) AS SELECT `test`.`tbl1`.`k1`, `test`.`tbl1`.`k2`\n" +
+                        "FROM `test`.`tbl1`;"
+        });
+        testCases.add(new String[]{"test_view_3",
+                "create view test_view_3 AS SELECT " +
+                        "*, `test`.`tbl1`.`k2` as k3 FROM `test`.`tbl1`",
+                "CREATE VIEW `test_view_3` (`k1`, `k2`, `v1`, `k3`) AS " +
+                        "SELECT `test`.`tbl1`.`k1`, `test`.`tbl1`.`k2`, `test`.`tbl1`.`v1`, `test`.`tbl1`.`k2` AS `k3`\n" +
+                        "FROM `test`.`tbl1`;"
+        });
+        testCases.add(new String[]{"test_view_4",
+                "create view test_view_4 AS " +
+                        "SELECT  `test`.`tbl1`.`k1` as c1, `test`.`tbl1`.`k2` as c2 FROM `test`.`tbl1`",
+                "CREATE VIEW `test_view_4` (`c1`, `c2`) AS SELECT `test`.`tbl1`.`k1` AS `c1`, `test`.`tbl1`.`k2` AS `c2`\n" +
+                        "FROM `test`.`tbl1`;"
+        });
+
+        ConnectContext ctx = starRocksAssert.getCtx();
+        for (String[] testcase : testCases) {
+            String dropViewSql = "drop view if exists " + testcase[0];
+            DropTableStmt dropViewStmt = (DropTableStmt) UtFrameUtils.parseStmtWithNewParser(dropViewSql, ctx);
+            GlobalStateMgr.getCurrentState().dropTable(dropViewStmt);
+            CreateViewStmt createViewStmt = (CreateViewStmt) UtFrameUtils.parseStmtWithNewParser(testcase[1], ctx);
+            GlobalStateMgr.getCurrentState().createView(createViewStmt);
+
+            List<Table> views = GlobalStateMgr.getCurrentState().getDb(createViewStmt.getDbName()).getViews();
+            List<String> res = Lists.newArrayList();
+            GlobalStateMgr.getDdlStmt(createViewStmt.getDbName(), views.get(0), res,
+                    null, null, false, false);
+
+            Assert.assertEquals(testcase[2], res.get(0));
+
+            GlobalStateMgr.getCurrentState().getDb(createViewStmt.getDbName()).dropTable(createViewStmt.getTable());
         }
     }
 

--- a/fe/fe-core/src/test/java/com/starrocks/analysis/ShowCreateViewStmtTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/analysis/ShowCreateViewStmtTest.java
@@ -147,6 +147,13 @@ public class ShowCreateViewStmtTest {
     @Test
     public void testCreateView() throws Exception {
         List<String[]> testCases = new ArrayList<>();
+        testCases.add(new String[]{"test_view_0",
+                "create view test_view_0 AS SELECT " +
+                        " *, concat('', null) FROM `test`.`tbl1`",
+                "CREATE VIEW `test_view_0` (`k1`, `k2`, `v1`, `concat('', NULL)`) " +
+                        "AS SELECT `test`.`tbl1`.`k1`, `test`.`tbl1`.`k2`, `test`.`tbl1`.`v1`, concat('', NULL)\n" +
+                        "FROM `test`.`tbl1`;"
+        });
         testCases.add(new String[]{"test_view_1",
                 "create view test_view_1 AS SELECT " +
                         "concat(`test`.`tbl1`.`k1`, `test`.`tbl1`.`k2`) FROM `test`.`tbl1`",

